Lib/ufo2ft/fastUFO.py

```diff
@@ -192,32 +247,71 @@ def __init__(
             self.groups = reader.readGroups()
             self.kerning = reader.readKerning()
             self.lib = reader.readLib()
-            glyphset = reader.getGlyphSet()
-            for name in glyphset.keys():
-                glyph = Glyph(name, self)
-                glyphset.readGlyph(
-                    glyphName=name, glyphObject=glyph, pointPen=glyph)
-                self.glyphset[name] = glyph
+            defaultLayerName = reader.getDefaultLayerName()
+            for layerName in reader.getLayerNames():
+                glyphset = reader.getGlyphSet(layerName)
+                if layerName not in self.layers:
+                    self.newLayer(layerName)
+                for name in glyphset.keys():
+                    glyph = self.layers[layerName].newGlyph(name)
+                    glyphset.readGlyph(
+                        glyphName=name, glyphObject=glyph, pointPen=glyph)
+            self._layers.defaultLayer = self._layers[defaultLayerName]
+            self.data.fileNames = reader.getDataDirectoryListing()
+        if self._layers.defaultLayer is None:
+            layer = self.newLayer(DEFAULT_LAYER_NAME)
+            self._layers.defaultLayer = layer
 
         self.kerningGroupConversionRenameMaps = None
 
     def __contains__(self, name):
-        return name in self.glyphset
+        return name in self._layers.defaultLayer
 
     def __getitem__(self, name):
-        return self.glyphset[name]
+        return self.layers.defaultLayer[name]
```

Comment:
self[name] should be made equivalent to self.layers.defaultLayer[name] as in defcon. defcon does that by returning self.layers.defaultLayer in self.glyphSet

Reply:
> self[name] should be made equivalent to self.layers.defaultLayer[name]

that's what this already does...

Reply:
Haha my bad I got confused, I meant to say that self.glyphSet instead of being deleted could be an alias of self.layers.defaultLayer as in defcon (it gets repeated a lot within the file)